brvs-core/src/main/java/iroha/validation/rest/RestService.java

```diff
@@ -108,15 +147,113 @@ public Response getTransactions() throws InvalidProtocolBufferException {
   @Path("/query")
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Response query(String query) {
+  public Response sendQuery(String query) {
+    return sendQuery(query, false);
+  }
+
+  @POST
+  @Path("/query/sign")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response signQueryWithSigning(String query) {
+    return sendQuery(query, true);
+  }
+
+  private Response sendQuery(String query, boolean sign) {
     try {
       final Query.Builder builder = Query.newBuilder();
       parser.merge(query, builder);
-      return Response.status(200).entity(printer.print(irohaAPI.query(builder.build()))).build();
-    } catch (InvalidProtocolBufferException e) {
+      Query queryToSend = builder.build();
+      if (sign) {
+        queryToSend = new jp.co.soramitsu.iroha.java.Query(queryToSend)
+            .buildSigned(brvsAccountKeyPair);
+      }
+      if (!queryToSend.hasSignature()) {
+        final String msg = "Query does not have signature";
+        logger.error(msg);
+        throw new IllegalArgumentException(msg);
+      }
+      return Response.status(200).entity(printer.print(irohaAPI.query(queryToSend))).build();
+    } catch (InvalidProtocolBufferException | IllegalArgumentException e) {
       return Response.status(422).build();
     } catch (Throwable t) {
       return Response.status(500).build();
     }
   }
+
+  @POST
+  @Path("/batch")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response sendTransactionsBatch(String transactionList) {
+    return sendTransactionsBatch(transactionList, false);
```

Comment:
I suggest you create another method with the same functionality without a boolean argument.